libcontainer/cgroups/fs/cpu.go

```diff
@@ -20,15 +20,55 @@ func (s *CpuGroup) Name() string {
 }
 
 func (s *CpuGroup) Apply(d *cgroupData) error {
+	path, err := d.path("cpu")
+	if err != nil && !cgroups.IsNotFound(err) {
+		return err
+	}
+
+	// We should set the real-Time group scheduling settings before moving
+	// in the process because if the process is already in SCHED_RR mode
+	// adding it will fail (due to 'cpu.rt_runtime_us' having an initial
+	// value of 0 and it isn't inherited)
+	if d.config.Resources.CpuRtPeriod != 0 || d.config.Resources.CpuRtRuntime != 0 {
+		if path != "" {
+			if err := os.MkdirAll(path, 0755); err != nil {
+				return err
+			}
+		}
+		if err := s.SetRtSched(path, d.config); err != nil {
+			return err
+		}
+	}
+
+	defer func() {
+		if err != nil {
+			os.RemoveAll(path)
+		}
+	}()
```

Comment:
we should not need the defer here as the path will be removed if there's any errors.